mu4e/mu4e-draft.el

```diff
@@ -576,16 +578,6 @@ PARENT is the parent message, if any."
 (defun mu4e--prepare-draft-buffer (compose-type parent)
   "Prepare the current buffer as a draft-buffer.
 COMPOSE-TYPE and PARENT are as in `mu4e--draft'."
-  (cl-assert (member compose-type '(reply forward edit new)))
-  (cl-assert (eq (if parent t nil)
-                 (if (member compose-type '(reply forward)) t nil)))
-  ;; remember some variables, e.g for user hooks. These are permanent-local
-  ;; hence survive the mode-switch below (we do this so these useful vars are
-  ;; available in mode-hooks.
-  (setq-local
-   mu4e-compose-parent-message parent
-   mu4e-compose-type compose-type)
-
```

Comment:
I’ve cut this part in 2 block:

1. move the cl-assert to the `mu4e--prepare-draft` function (to keep the semantic of "preparing" stuff, ensuring everything is ok)
2. move the setq-local statement later in the `mu4e--draft` function to still benefit from being call inside the target buffer (as now they are buffer-local, and `mu4e--prepare-draft` is called before the buffer is created.